### ISSUE TITLE
Add a web application manifest file

### ISSUE THREAD

**Issue by alrra** [h5bp/html5boilerplate.com#133]
See:
- https://w3c.github.io/manifest/
- https://github.com/h5bp/html5-boilerplate/pull/1629

**Comment by arthurvr** [comment]
Added this to https://github.com/h5bp/html5boilerplate.com-mobile/issues/9 too. PITA we have to do everything twice.

**Comment by alexgibson** [comment]
Genuine query - but are manifest files not more to facilitate installation of web apps? (the common current use case being on mobile). Why add one to this site, what are the specific advantages here?

This seems like a perfect fit for Mobile Boilerplate, which is more web-app specific. But should every h5bp site (and also this website specifically) need this by default?

**Comment by alrra** [comment]
> what are the specific advantages here?

@alexgibson Basically for the _"Add to homescreen"_ functionality (which is available in [Chrome from Android](https://developer.chrome.com/multidevice/android/installtohomescreen#manifestlink), but [also coming to Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=997779)). The basic idea is that if we add the `apple-touch-icon` we should add this too, more as an example then anything else.

**Comment by alexgibson** [comment]
Firefox has already supported Open Web App manifest installation for some time, so it's great to see work standardising between two browsers here :+1: 

Apple touch icons are more akin to favicons on iOS (since any site can be bookmarked on the home screen by default). But I think simply bookmarking sites in this fashion is maybe an over simplification of what manifests serve to be.

Just my 2 cents :)

**Comment by alexgibson** [comment]
Although I do see advantages in removing all that clutter in the `<head>` :smile:

**Comment by alrra** [comment]
> Apple touch icons are more akin to favicons on iOS (since any site can be bookmarked on the home screen by default). But I think simply bookmarking sites in this fashion is maybe an over simplification of what manifests serve to be.

I agree that it is oversimplification, but Chrome kinda when from using the `apple-touch-icon`s to using (prior to `v39`) [their own (?) syntax](https://developer.chrome.com/multidevice/android/installtohomescreen#icon) to using the web application manifest file. Firefox will do the same, and maybe then some of the [other browsers](https://status.modern.ie/webapplicationmanifest) will follow. 

I think (and feel free to disagree) that, providing a __starting__ web application manifest file is in general a good idea, even if it will not be used to its "full potential". Having a file that is requested only when needed, that basically contains the metadata for the  site/web app, it is a better way of doing things then [cluttering](https://cloud.githubusercontent.com/assets/1223565/6163944/62967ea8-b2a1-11e4-8f8d-7a7230ea131c.png) the `<head>` of every page.

**Comment by alrra** [comment]
@alexgibson Sorry I missed your second comment .

**Comment by alexgibson** [comment]
Some good points, I'm not strongly disagreeing with adding this, just curious as to the use cases for websites that don't require the more advanced features. Only thing I think the web needs to avoid `apple-web-app-capable` scenario where site publishers added it without really testing behavior thoroughly.

**Comment by alrra** [comment]
> Only thing I think the web needs to avoid apple-web-app-capable scenario where site publishers added it without really testing behavior thoroughly.

@alexgibson The only things H5BP can do to prevent that sort of issues is try to [figure out](https://github.com/h5bp/html5-boilerplate/pull/1629) a good default that is not harmful for anyone, and try to educate / point users to the right learning resources. Unfortunately, other then that, you can't really do anything to ensure that users will not do the wrong things

**Comment by alrra** [comment]
This morning I wanted to close this issue, but it seems that there might be a bug in [Chrome for Android](https://cloud.githubusercontent.com/assets/1223565/7411442/b339b90c-ef40-11e4-8c0b-a3bf4a1fabda.png) (Blink) related to the CSP header:

e.g.: 

* If `test.html` contains `<link rel="manifest" href="site.webmanifest">`, and is served with the following CSP header:

   ```http
Content-Security-Policy: "default-src 'none';manifest-src 'self';report-uri http://192.168.2.105:8080/"
```
  
    Chrome won't request the manifest file when the user taps on the `Add to Home screen` menu item as shown also by the CSP report:

   ```
{ 'csp-report': 
   { 'document-uri': 'http://192.168.2.105/',
     referrer: '',
     'violated-directive': 'default-src \'none\'',
     'effective-directive': 'manifest-src',
     'original-policy': 'default-src \'none\';manifest-src \'self\';report-uri http://192.168.2.105:8080/',
     'blocked-uri': 'http://192.168.2.105/site.webmanifest',
     'status-code': 200 } }
```

   The same happens if the CSP header is:

   ```http
Content-Security-Policy: "default-src 'none';manifest-src *;report-uri http://192.168.2.105:8080/"
```

   So, basically, the value specified by [`manifest-src`](https://w3c.github.io/webappsec/specs/content-security-policy/#directive-manifest-src) directive doesn't overwrite the value specified by `default-src`.

Cc @marcoscaceres, @mikewest

--

Note: Didn't find an open issue on this (have to check again when I have more time), but if I haven't missed something from the specification, this seems like a bug to me.

**Comment by paulirish** [comment]
@alrra sounds like a bug! can you file on crbug.com ?

**Comment by marcoscaceres** [comment]
@alrra, @paulirish, according to @kenchris, it's not enabled yet. @kenchris, can you provide details please.

**Comment by kenchris** [comment]
Yes, Chrome supports it, but as it wasn't in any spec at the point that I implemented it, it was put behind a flag. I guess an intent to ship needs to be send and we can enable it.

**Comment by roblarsen** [comment]
we added one, I've just been punch drunk since the release.
